lib/customisation.nix

```diff
@@ -287,7 +320,25 @@ rec {
     { inherit otherSplices keep extra f; };
 
   /* Like makeScope, but aims to support cross compilation. It's still ugly, but
-     hopefully it helps a little bit. */
+     hopefully it helps a little bit.
+
+     Type:
+       makeScopeWithSplicing' ::
+         { splicePackages :: Splice -> AttrSet
+         , newScope :: AttrSet -> (AttrSet -> A | Path) -> AttrSet -> A
+         }
+         -> { otherSplices :: Splice, keep :: AttrSet -> AttrSet, extra :: AttrSet -> AttrSet }
+         -> AttrSet
+
+       Splice ::
+         { pkgsBuildBuild :: AttrSet
+         , pkgsBuildHost :: AttrSet
+         , pkgsBuildTarget :: AttrSet
+         , pkgsHostHost :: AttrSet
+         , pkgsHostTarget :: AttrSet
+         , pkgsTargetTarget :: AttrSet
+         }
```

Comment:
Will this render to something that makes sense? To be clear, it's not wrong to do it that way, but if it ends up as a messy pile of characters we may have to reconsider how to present it.

Reply:
It renders to a multiple-line code block just like most of the `Example` blocks, at least when viewing the rendered HTML file with Firefox.

This is the first time I use multiple-line `Type` block. I'm not sure if it's the proper representation, but it seems to be more clearly than a super-long single line.